lib/stitches/auto_fill_test.py

```diff
@@ -0,0 +1,200 @@
+from shapely import geometry as shgeo
+
+from lib.stitches import auto_fill
+from lib.stitches.utils import testing
+from lib.stitch_plan.stitch import Stitch
+
+
+def test_auto_fill_box():
+    # a box with side length 10 and lower-left corner at (10, 10)
+    shape = shgeo.box(10, 10, 20, 20, ccw=True)
+
+    actual = auto_fill(
+        shape=shape,
+        angle=0,
+        row_spacing=1,
+        end_row_spacing=1,
+        max_stitch_length=1,
+        running_stitch_length=1,
+        running_stitch_tolerance=1,
+        staggers=4,
+        skip_last=False,
+        starting_point=(10, 10),
+        ending_point=(10, 10),
+        underpath=False,
+    )
+
+    # the start and ending points should be the same
+    assert actual[0].as_point().isclose(actual[-1].as_point())
+
+    # the exact stitches output by the algorithm now
+    expected = [
+        Stitch(*p).as_point()
+        for p in [
+            (10.0, 10.0),
+            (11.0, 10.0),
+            (12.0, 10.0),
+            (13.0, 10.0),
+            (14.0, 10.0),
+            (15.0, 10.0),
+            (16.0, 10.0),
+            (17.0, 10.0),
+            (18.0, 10.0),
+            (19.0, 10.0),
+            (20.0, 10.0),
+            (20.0, 11.0),
+            (20.0, 12.0),
+            (20.0, 13.0),
+            (20.0, 14.0),
+            (20.0, 15.0),
+            (20.0, 16.0),
+            (20.0, 17.0),
+            (20.0, 18.0),
+            (20.0, 19.0),
+            (20.0, 20.0),
+            (20.0, 20.0),
+            (20.0, 20.0),
+            (19.0, 20.0),
+            (18.0, 20.0),
+            (17.0, 20.0),
+            (16.0, 20.0),
+            (15.0, 20.0),
+            (14.0, 20.0),
+            (13.0, 20.0),
+            (12.0, 20.0),
+            (11.0, 20.0),
+            (10.0, 20.0),
+            (10.0, 19.0),
+            (10.0, 19.0),
+            (10.75, 19.0),
+            (11.75, 19.0),
+            (12.75, 19.0),
+            (13.75, 19.0),
+            (14.75, 19.0),
+            (15.75, 19.0),
+            (16.75, 19.0),
+            (17.75, 19.0),
+            (18.75, 19.0),
+            (19.75, 19.0),
+            (20.0, 18.0),
```

Comment:
Is this sort of diagonal stitch expected?

Reply:
![image](https://user-images.githubusercontent.com/2467754/200998725-d855b4cb-a126-45b9-8ecd-ac414eb35d74.png)

This is the stitches converted to an SVG polyline. Not perfect, but you can see the diagonal stitches at the end of each row. My guess is that this is OK and expected, but I just wanted to confirm.

Reply:
Yup, that's okay and expected -- that's the "skip last stitch" setting, I think.  In practice, with multiple rows per millimeter, the diagonal isn't noticeable.